rhino/src/main/java/org/mozilla/javascript/NativeProxy.java

```diff
@@ -0,0 +1,1322 @@
+/* -*- Mode: java; tab-width: 4; indent-tabs-mode: 1; c-basic-offset: 4 -*-
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.javascript;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * This class implements the Proxy object.
+ *
+ * @author Ronald Brill
+ */
+final class NativeProxy extends ScriptableObject implements Callable, Constructable {
```

Comment:
I see in "Init" that you are creating a LambdaConstructor and putting in the scope, so that it will become the constructor called "Proxy" in the scope. But you also made this class Callable and Constructable, which you don't actually need to do if (and only if) that's the only constructor that you need -- in other lambda things, I've actually had the LambdaConstructor call a static function. See NativePromise for an example.

So given that, do you need this object to be Constructable and Callable for another reason?

Reply:
And given all that, I probably need to understand better what a Proxy object actually does so yes, you might need that -- just trying to understand it all!

Reply:
Having read NativeReflect I get it now! Thanks.